posthog/session_recordings/queries/test/__snapshots__/test_session_recording_list_from_session_replay.ambr

```diff
@@ -3677,6 +3486,56 @@
   OFFSET 0
   '''
 # ---
+# name: TestClickhouseSessionRecordingsListFromSessionReplay.test_person_id_filter_with_distinct_id_optimization
+  '''
+  
+  SELECT s.session_id,
+         any(s.team_id),
+         any(s.distinct_id),
+         min(s.min_first_timestamp) as start_time,
+         max(s.max_last_timestamp) as end_time,
+         dateDiff('SECOND', start_time, end_time) as duration,
+         argMinMerge(s.first_url) as first_url,
+         sum(s.click_count),
+         sum(s.keypress_count),
+         sum(s.mouse_activity_count),
+         sum(s.active_milliseconds)/1000 as active_seconds,
+         duration-active_seconds as inactive_seconds,
+         sum(s.console_log_count) as console_log_count,
+         sum(s.console_warn_count) as console_warn_count,
+         sum(s.console_error_count) as console_error_count
+  FROM session_replay_events s
+  WHERE s.team_id = 2
+    AND s.min_first_timestamp >= '2020-12-11 13:46:23'
+    AND s.min_first_timestamp >= '2020-12-25 00:00:00'
+    AND s.max_last_timestamp <= '2021-01-01 13:46:23'
+    AND s.distinct_id in
+      (select distinct_id
+       from
+         (SELECT distinct_id,
+                 argMax(person_id, version) as person_id
+          FROM person_distinct_id2 as pdi
+          WHERE team_id = 2
+            AND distinct_id in
+              (SELECT distinct_id
+               FROM events e
+               WHERE team_id = 2
+                 AND e.timestamp >= '2020-12-11 13:46:23'
+                 AND e.timestamp <= now()
+                 AND timestamp >= '2020-12-24 12:00:00'
+                 AND timestamp <= '2021-01-02 01:46:23'
+                 and $session_id != ''
+                 and event = '$pageview') as load_fewer_rows
+          GROUP BY distinct_id
+          HAVING argMax(is_deleted, version) = 0
+          and person_id = '00000000-0000-0000-0000-000000000000') as session_persons_sub_query)
```

Comment:
here's the meat of the change...

now we check for distinct ids that might have sessions in the time range and limit our search against persons for those distinct ids

it's not perfect but does help when tested in metabase